src/scripts/main.js

```diff
@@ -1 +1,83 @@
 'use strict';
+
+let isPromise1Clicked = false;
+let isLeftMbClicked = false;
+let isRightMbClicked = false;
+
+const promise1 = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    if (!isPromise1Clicked) {
+      reject(new Error('First promise was rejected'));
+    }
+  }, 3000);
+
+  document.addEventListener('click', () => {
+    isPromise1Clicked = true;
+    clearTimeout(timeoutId);
+    resolve();
+  });
+});
+
+promise1.then(() => {
+  const div = document.createElement('div');
+
+  div.innerHTML = 'First promise was resolved';
+  div.className = 'success';
+  div.setAttribute('data-qa', 'notification');
+  document.body.append(div);
+});
+
+promise1.catch(() => {
+  const div = document.createElement('div');
+
+  div.innerHTML = 'First promise was rejected';
+  div.className = 'error';
+  div.setAttribute('data-qa', 'notification');
+  document.body.append(div);
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  document.addEventListener('contextmenu', (e) => {
+    e.preventDefault();
+    resolve();
+  });
+
+  document.addEventListener('click', () => {
+    resolve();
+  });
```

Comment:
The current implementation of `promise2` resolves on both right-click (contextmenu) and left-click events. If the task requires resolving only on a specific type of click, modify the logic to meet this requirement.